Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -1166,10 +1177,44 @@ WarpXOpenPMDPlot::GetMeshCompNames (int meshLevel,
     }
 
     if ( 0 == meshLevel )
-      return;
+        return;
 
     field_name += std::string("_lvl").append(std::to_string(meshLevel));
 }
+/* Split varname into name and theta mode mode index
+ *
+ * @param varname [IN]: variable name
+ * @param varname field_str_end_index [OUT] : if varname = fieldName_modeNumber_realOrImag,  index of underscore after fieldName, otherwise, varname.size()
+ * @param varname mode_index [OUT] : if varname = fieldName_modeNumber_realOrImag, returns 2 * modeNumber - 1 + (realOrImag == 'imag'), otherwise, -1
+ */
+std::tuple<int, int>
+GetFieldModeIndices (const std::string& varname)
+{
+    int mode_index = -1;
+    int field_str_end_index = varname.size();
+    if (varname.size() >= 6) {
+        std::string real_imag = varname.substr(field_str_end_index - 4);
+        if (real_imag == "real" || real_imag == "imag") {
+            if (real_imag[0] == 'i') {
+                mode_index += 1;
+            }
+            field_str_end_index -= 6; // varname has form <varname>_modeNumber_realOrImag
+                                      // extracting modeNumber:
+            int num_mode_digits = 0;
+            while ( varname[field_str_end_index] != '_') {
+                field_str_end_index--;
+                num_mode_digits++;
+            }
+            int mode = std::stoi(varname.substr(field_str_end_index+1, num_mode_digits));
+            if (mode == 0) {
+                mode_index = 0;
+            } else {
+                mode_index += 2 * mode;
```

Comment:
+1?

I have to admit, I cannot follow the logic here yet, need more inline comments on what every part does.